### ISSUE TITLE
Overlapping relic nodes do not give points

### ISSUE THREAD

**Issue by w9PcJLyb** [Lux-AI-Challenge/Lux-Design-S3#14]
For example, with seed 6244306, the tile (14, 9) does not give points

<img width="631" alt="Screenshot_20241017_100424" src="https://github.com/user-attachments/assets/d2406aff-c147-48e0-bc10-edc3a70f6d75">

----

I think the `team_relic_score' function should look like this:

```python
# Compute relic scores
def team_relic_score(unit_counts_map):
    scores = (unit_counts_map > 0) & (state.relic_nodes_map_weights > 0)
    return jnp.sum(scores, dtype=jnp.int32)
```
